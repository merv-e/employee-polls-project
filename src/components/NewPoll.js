
const NewPoll = (props) => {

    const handleSubmit = (e) => {  
    e.preventDefault();
    //Todo : to be built.. 
    };

  return (
    <div className="poll">
        <h2 className="center">Would you rather</h2>
        <form className="poll" onSubmit={(e) => handleSubmit(e)}>
          <input placeholder="First Option"/>
          <input placeholder="Second Option"/>
          <button className="btn" onClick={handleSubmit}>Submit</button>
        </form>
    </div>
  )
}

export default NewPoll
