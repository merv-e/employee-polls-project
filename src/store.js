import { configureStore } from '@reduxjs/toolkit';
import users from "./reducers/users";
import questions from './reducers/questions';
import { loadingBarReducer } from 'react-redux-loading-bar';
import authedUser from './reducers/authedUser';

export const store = configureStore({
  reducer: {
    users,
    questions,
    authedUser, 
    loadingBar : loadingBarReducer  
  }
});