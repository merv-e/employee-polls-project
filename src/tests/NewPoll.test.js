import { render, screen } from "@testing-library/react";
import *as React from "react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import NewPoll from "../components/NewPoll";
// import store from "../index"; //store.js'den al :)
import {store} from "../store";

// test("should have the expected form elements", ()=> {
//         const utils = render(
//             // <Provider store={store}>
//             // <MemoryRouter>
//                     <NewPoll />
//             // </MemoryRouter>
//                 // </Provider>
//         );
        
//         // screen.
//         expect(utils).toMatchSnapshot();
//         // expect(getByTestId("first-option-input")).toBeInTheDocument();
        
//         // const fOption = screen.getByTestId("first-option-input")
//         // expect(fOption).toBeInTheDocument();
// });


// describe("NewPoll", ()=> {
//     it('should have the expected form elements', () => {
//         const utils = render(
//             // <MemoryRouter>
//             //     <Provider store={store}>
//                     <NewPoll />
//             //     </Provider>
//             // </MemoryRouter>
//         );
        
//         // screen.
//         expect(utils).toMatchSnapshot();
//         // expect(getByTestId("first-option-input")).toBeInTheDocument();
        
//         // const fOption = screen.getByTestId("first-option-input")
//         // expect(fOption).toBeInTheDocument();
//     });
// });