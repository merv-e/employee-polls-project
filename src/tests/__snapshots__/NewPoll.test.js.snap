// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPoll should take a screenshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form-poll"
      >
        <h2>
          Would you rather
        </h2>
        <form
          class="form-poll"
        >
          <label
            for="fOption"
          >
            First Option
          </label>
          <input
            data-testid="first-option-input"
            name="fOption"
            placeholder="Please write down your first option"
            value=""
          />
          <label
            for="secOption"
          >
            Second Option
          </label>
          <input
            data-testid="second-option-input"
            name="secOption"
            placeholder="Please write down your second option"
            value=""
          />
          <button
            class="btn"
            data-testid="submitBtn"
            disabled=""
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form-poll"
    >
      <h2>
        Would you rather
      </h2>
      <form
        class="form-poll"
      >
        <label
          for="fOption"
        >
          First Option
        </label>
        <input
          data-testid="first-option-input"
          name="fOption"
          placeholder="Please write down your first option"
          value=""
        />
        <label
          for="secOption"
        >
          Second Option
        </label>
        <input
          data-testid="second-option-input"
          name="secOption"
          placeholder="Please write down your second option"
          value=""
        />
        <button
          class="btn"
          data-testid="submitBtn"
          disabled=""
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
